Components/ProductToShop.php

```diff
@@ -511,15 +511,19 @@ private function applyProductProperties(ProductModel $article, Product $product)
             $this->manager->flush();
         }
 
+        $propertyValues = $article->getPropertyValues();
+        $propertyValues->clear();
+        $this->manager->persist($article);
+        $this->manager->flush();
+
         $article->setPropertyGroup($group);
 
-        $valueCollection = new ArrayCollection();
         $optionRepo = $this->manager->getRepository(PropertyOption::class);
         $valueRepo = $this->manager->getRepository(PropertyValue::class);
 
         foreach ($product->properties as $property) {
             $option = $optionRepo->findOneBy(['name' => $property->option]);
-
+            $optionExists = $option instanceof PropertyOption;
```

Comment:
Imho it is a code smell if there are too many temporary variables like this. I would prefer to write it directly into the `if` statement. Anyway, you get my approve it is just a little thing (over time it gets messy. files could save ~100 lines of code or more sometimes ;) )